### PR TITLE
Drop support for Python 3.8 & 3.9, add support for Python 3.12 in CI/CD and update to latest packages

### PR DESCRIPTION
**Pull request by iMicknl**
This pull request includes multiple changes aimed at updating dependencies, improving type hints, and removing outdated configurations. The most important changes include updating the Python version, enhancing type hints, and cleaning up IntelliJ IDEA project files.

## Breaking

- Renamed private _ssl_context to _ssl: ssl.SSLContext
- Remove support for Python 3.8 and 3.9, bumping the minimum version to Python 3.10

### Dependency and Version Updates:
* [`.devcontainer/devcontainer.json`](diffhunk://#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686L6-L12): Updated the Docker image to `mcr.microsoft.com/devcontainers/python:1-3.12-bullseye` for Python 3.12.
* [`.github/workflows/main.yaml`](diffhunk://#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3L19-R23): Updated the Python versions in the matrix to include 3.12 and removed 3.8 and 3.9.
* [`.python-version`](diffhunk://#diff-652bad7c1002badcf13f8de3a5096c2724ce54ae36cd68a21ad280547dff8e87L1-R1): Updated the Python version to 3.12.
* [`pyproject.toml`](diffhunk://#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L16-R35): Updated dependencies and Python version requirements to `>=3.10,<4.0`.

### Type Hint Improvements:
* [`pyoverkiz/client.py`](diffhunk://#diff-11513003e65960c0b1a4bccb3c6bf2b7dea08c03923a8e53b8dea1a05f213aa2L117-L126): Changed `_ssl_context` to `_ssl` and updated type hints for SSL context and verification. [[1]](diffhunk://#diff-11513003e65960c0b1a4bccb3c6bf2b7dea08c03923a8e53b8dea1a05f213aa2L117-L126) [[2]](diffhunk://#diff-11513003e65960c0b1a4bccb3c6bf2b7dea08c03923a8e53b8dea1a05f213aa2R148-R160) [[3]](diffhunk://#diff-11513003e65960c0b1a4bccb3c6bf2b7dea08c03923a8e53b8dea1a05f213aa2L896-R898) [[4]](diffhunk://#diff-11513003e65960c0b1a4bccb3c6bf2b7dea08c03923a8e53b8dea1a05f213aa2L916-R918) [[5]](diffhunk://#diff-11513003e65960c0b1a4bccb3c6bf2b7dea08c03923a8e53b8dea1a05f213aa2L933-R935)
* [`pyoverkiz/models.py`](diffhunk://#diff-abae13362160dc5f0aff1ba910ffe223a87f52dfda3a77f160ad13a6a9dbc92cL4-R5): Updated imports to use `collections.abc` for `Iterator`.
* [`pyoverkiz/types.py`](diffhunk://#diff-338e5a2c49dfd3de0370d183f0e7525f931f9d2934d49c00270620a4f9d0f083L4-R9): Updated type hints to use `collections.abc` for `Callable` and modernized `StateType` and `JSON` definitions. [[1]](diffhunk://#diff-338e5a2c49dfd3de0370d183f0e7525f931f9d2934d49c00270620a4f9d0f083L4-R9) [[2]](diffhunk://#diff-338e5a2c49dfd3de0370d183f0e7525f931f9d2934d49c00270620a4f9d0f083L20-R21)

### Configuration Cleanup:
* Removed IntelliJ IDEA project files including `.idea/inspectionProfiles/Project_Default.xml`, `.idea/inspectionProfiles/profiles_settings.xml`, `.idea/misc.xml`, `.idea/modules.xml`, `.idea/python-tahoma-api.iml`, and `.idea/vcs.xml`. [[1]](diffhunk://#diff-bbf0d02ebcf6764cae92fa4045a2d6f15f8a479cd1aa2ab4d1fd435a3aa74bbdL1-L25) [[2]](diffhunk://#diff-9fb34448452829ed3cd715b49c0f4494fe0b97a4aef2a5b23e12c486ed932273L1-L6) [[3]](diffhunk://#diff-7ea1cd7bd5d16299a0abd33128e8d0a8bc04146f6b00c07bcd3cc0aad813c229L1-L4) [[4]](diffhunk://#diff-0aae258f1732eac2acd4fcdd1af6d0737b4ec8d233136c3ba7c40b6d49aeda50L1-L8) [[5]](diffhunk://#diff-232d54012a1560d27c50b6224fd355aa68a8e0593a849acfe1dd446aa9f1a3aaL1-L20) [[6]](diffhunk://#diff-a6dab70bd0e9ffc91d8419ad52225baff3a3e7c9d8e1724f7ed6d96b661d804aL1-L6)

### Pre-commit and Tox Configuration:
* [`.pre-commit-config.yaml`](diffhunk://#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9L3-R3): Updated `pre-commit-hooks` to `v4.6.0` and changed `pyupgrade` arguments to `--py310-plus`. [[1]](diffhunk://#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9L3-R3) [[2]](diffhunk://#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9L30-R30)
* [`tox.ini`](diffhunk://#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L3-R3): Updated `envlist` to include Python 3.12 and removed 3.8 and 3.9. [[1]](diffhunk://#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L3-R3) [[2]](diffhunk://#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L14-R24)

### Miscellaneous:
* [`.github/CODEOWNERS`](diffhunk://#diff-3d36a1bf06148bc6ba1ce2ed3d19de32ea708d955fed212c0d27c536f0bd4da7L1-R1): Removed `@vlebourl` from the list of code owners.
* [`.github/release-drafter.yml`](diffhunk://#diff-101bec72d0f1f84e7290d113531bbd8c6aa355cdc9f456df255544bc0a2da0eaR34): Added an empty line for better formatting.
* [`tests/test_client.py`](diffhunk://#diff-0d92063e88430a02df61616c5f16b148b64ac4539d9cb9b8d883d5a23351b110L9-R9): Updated fixtures to be asynchronous. [[1]](diffhunk://#diff-0d92063e88430a02df61616c5f16b148b64ac4539d9cb9b8d883d5a23351b110L9-R9) [[2]](diffhunk://#diff-0d92063e88430a02df61616c5f16b148b64ac4539d9cb9b8d883d5a23351b110L23-R27)

### CODE REVIEW

**Comment by iMicknl** [comment]
@tetienne quite a big PR in the end. If you would have time coming week to have a quick glance, would be appreciated. I will do some tests on HA with this package, since there have been some (potential) breaking changes.

**Comment by tetienne** [comment]
> @tetienne quite a big PR in the end. If you would have time coming week to have a quick glance, would be appreciated. I will do some tests on HA with this package, since there have been some (potential) breaking changes.

Sure I will have a look. From my phone, it looks almost good.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,6 +9,11 @@ on:
 jobs:
   deploy:
     runs-on: ubuntu-latest
+    environment: release
```
Comment (tetienne) [inline .github/workflows/release.yml:12]: We are not using the environment. We can remove this IMO.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -31,10 +36,8 @@ jobs:
           ref: "main"
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build and publish to PyPi
-        env:
-          POETRY_PYPI_TOKEN_PYPI: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (tetienne) [inline .github/workflows/release.yml:36]: Once the new release process confirmed, don't forget to remove the secret.
By the way, I really like this news trusted publishing :)

File: pyoverkiz/client.py [hunk 3]
```diff
@@ -114,16 +114,16 @@ class OverkizClient:
     _refresh_token: str | None = None
     _expires_in: datetime.datetime | None = None
     _access_token: str | None = None
-    _ssl_context: ssl.SSLContext | None = None
+    _ssl: ssl.SSLContext | bool = True
```
Comment (tetienne) [inline pyoverkiz/client.py:117]: Should we include this as breaking change?

**Review by tetienne — APPROVED** [review]
Looks good to me. To limit the risk, you can also split this PR into small ones.

File: .github/workflows/release.yml (hunk 1 above)
Comment (iMicknl) [reply .github/workflows/release.yml:12]: @tetienne we need this actually for the connection to PyPi. They are tied to an environment.

File: pyoverkiz/client.py (hunk 3 above)
Comment (iMicknl) [reply pyoverkiz/client.py:117]: It is a private setter so technically not a breaking change, but good to mention it. Here I am most afraid of breaking anything by the way, so will need to do proper testing before adding to HA.

File: .github/workflows/release.yml (hunk 2 above)
Comment (iMicknl) [reply .github/workflows/release.yml:36]: Good one!

**Pull request merged**